### PR TITLE
Support CMIP3

### LINKED ISSUE #27: Read in CMIP3 datasets
I need to read in **CMIP3** datasets in the ESMValTool. I started to implement this feature in the code but it was not successful as I did something wrong or missed something for sure.
Is there anyone who could support me?

What I did:
I modified `cmor/table.py` and `config-developer.yml` and added the CMOR table file `cmor/tables/cmip3/Tables/CMIP3_A1` (see branch `version2_development_cmip3`).
And I tried it with adding this dataset lines in the `recipes/recipe_python.yml`:
```
  - {dataset: ukmo_hadcm3,  project: CMIP3,  mip: A1,  exp: 20c3m,  ensemble: run1,  start_year: 1997,  end_year: 1999}
  - {dataset: ukmo_hadgem1,  project: CMIP3,  mip: A1,  exp: 20c3m,  ensemble: run1,  start_year: 1997,  end_year: 1999}
```
On DKRZ the data structure is like this:
`<CMIP3-directory>/20c3m/atm/mo/pr/ukmo_hadgem1/run1/pr_A1.nc`

The error message is:
`esmvaltool._recipe_checks.RecipeError: Unable to load CMOR table 'CMIP3' for variable 'tas' with mip 'A1'`

### PR DESCRIPTION
**Pull request by jvegasbsc**
This pull request aims to add support for CMIP3, which is required for comparison works between different epochs of the CMIP project.

Working with @LisaBock 

Closes #27

### CODE REVIEW

**Comment by jvegasbsc** [comment]
I have updated the branch in ESMValTool to keep the recipe and ported all the changes here.

I can not test with data because I don't have access to CMIP3 in Jasmin. I already requested it, but I don't know when I will be able to use it

@LisaBock, can you give it a try?

**Comment by jvegasbsc** [comment]
One remark: mip selection in CMIP3 does not have information about the frequency. It must be added manually

**Comment by LisaBock** [comment]
I tested it, but it stopped here:
```
2019-09-24 09:49:47,592 UTC [40340] INFO    Running 2 tasks using 2 processes
2019-09-24 09:49:47,700 UTC [46884] INFO    Starting task diagnostic1/tas in process [46884]
2019-09-24 09:49:47,706 UTC [46884] DEBUG   Running block ['fix_metadata', 'concatenate', 'cmor_check_metadata', 'extract_time', 'fix_data', 'regrid', 'cmor_check_data']
2019-09-24 09:49:47,706 UTC [46884] DEBUG   Applying single-model steps to PreprocessorFile: /scratch/b/b309059/esmvaltool_output/recipe_cmip3_20190924_094941/preproc/diagnostic1/tas/CMIP3_ncar_ccsm3_0_A1_20c3m_run1_tas_1997-1999.nc
2019-09-24 09:49:47,707 UTC [46884] DEBUG   Running preprocessor step fix_file
2019-09-24 09:49:47,707 UTC [46884] DEBUG   Running fix_file(/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_ccsm3_0/run1/tas_A1.20C3M_1.CCSM.atmm.1990-01_cat_1999-12.nc, {'project': 'CMIP3', 'dataset': 'ncar_ccsm3_0', 'short_name': 'tas', 'output_dir': '/scratch/b/b309059/esmvaltool_output/recipe_cmip3_20190924_094941/preproc/diagnostic1/tas/CMIP3_ncar_ccsm3_0_A1_20c3m_run1_tas_1997-1999_fixed'})
2019-09-24 09:49:47,708 UTC [46884] DEBUG   Running preprocessor step load
2019-09-24 09:49:47,708 UTC [46884] DEBUG   Running load(/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_ccsm3_0/run1/tas_A1.20C3M_1.CCSM.atmm.1990-01_cat_1999-12.nc, {'callback': <function concatenate_callback at 0x2ad99e9ebb70>})
2019-09-24 09:49:47,708 UTC [46884] DEBUG   Loading:
/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_ccsm3_0/run1/tas_A1.20C3M_1.CCSM.atmm.1990-01_cat_1999-12.nc
2019-09-24 09:49:47,750 UTC [46884] DEBUG   Running preprocessor step fix_metadata
2019-09-24 09:49:47,751 UTC [46884] DEBUG   Running fix_metadata([<iris 'Cube' of air_temperature / (K) (time: 120; latitude: 128; longitude: 256)>], {'project': 'CMIP3', 'dataset': 'ncar_ccsm3_0', 'short_name': 'tas', 'cmor_table': 'CMIP3', 'mip': 'A1', 'frequency': 'mo'})
2019-09-24 09:49:47,790 UTC [40340] INFO    Progress: 1 tasks running, 1 tasks waiting for ancestors, 0/2 done
2019-09-24 09:49:47,940 UTC [46884] WARNING There were warnings in variable tas:
Long name for tas changed from air_temperature to Surface Air Temperature
```
I waited now about 30 minutes but nothing happens anymore.

**Comment by jvegasbsc** [comment]
Thanks Lisa. I now have access to CMIP3 data in JASMIN. I will take a look and let you know

**Comment by jvegasbsc** [comment]
Tested in Jasmin with this recipe for `tas` and `zg`. Working nicely
```yaml
datasets:
  - &cmip3 {project: CMIP3, exp: 20c3m, dataset: BCM2,  ensemble: r1, start_year: 1979, end_year: 1990}
  - {<<: *cmip3, dataset: CGCM3-1-T47}
  - {<<: *cmip3, dataset: CGCM3-1-T63}
  - {<<: *cmip3, dataset: BCC-CM1}
  - {<<: *cmip3, dataset: CM3, institute: INM}
  - {<<: *cmip3, dataset: CM3, institute: CNRM}
  - {<<: *cmip3, dataset: MK3}
  - {<<: *cmip3, dataset: MK3-5}
  - {<<: *cmip3, dataset: CM2}
  - {<<: *cmip3, dataset: CM2-1}
  - {<<: *cmip3, dataset: ECHAM4}
  - {<<: *cmip3, dataset: CM4}
  - {<<: *cmip3, dataset: FGOALS-G1-0}
  - {<<: *cmip3, dataset: ECHO-G}
  - {<<: *cmip3, dataset: ECHAM5}
  - {<<: *cmip3, dataset: GISS-AOM}
  - {<<: *cmip3, dataset: GISS-EH}
  - {<<: *cmip3, dataset: GISS-ER}
  - {<<: *cmip3, dataset: CCSM3}
  - {<<: *cmip3, dataset: PCM}
  - {<<: *cmip3, dataset: MIROC3-2-HI}
  - {<<: *cmip3, dataset: MIROC3-2-MED}
  - {<<: *cmip3, dataset: HADCM3}
  - {<<: *cmip3, dataset: HADGEM1}

preprocessors:
  clim:
    climate_statistics:                                                                                                                                                                                                                                                        r      period: 'season'
                                                                                                                                                                                                                                                                               Cdiagnostics:
  sit_anom:
    description: Computes the negative seaice feedback                                                                                                                                                                                                                         2    variables:
      zg:
        mip: A1
        frequency: mon
        modeling_realm: atmos
        preprocessor: clim
    scripts: null
```

@LisaBock, can you test the same on DKRZ and let me know what happens?

**Comment by LisaBock** [comment]
The version of our CMIP3 data seems to be a different one. The names of our models are 
```
bccr_bcm2_0    cccma_cgcm3_1_t63  gfdl_cm2_0  giss_aom        giss_model_e_r   ingv_echam4  ipsl_cm4        miroc3_2_medres  mpi_echam5      ncar_ccsm3_0  ukmo_hadcm3
cccma_cgcm3_1  csiro_mk3_0        gfdl_cm2_1  giss_model_e_h  iap_fgoals1_0_g  inmcm3_0     miroc3_2_hires  miub_echo_g      mri_cgcm2_3_2a  ncar_pcm1     ukmo_hadgem1
```
I think it is not that big problem but I can't use the institute table you defined in `config_developer.yml`.

**Comment by LisaBock** [comment]
`zg` works fine at DKRZ. I tried it with `tas` but it stopped for different models here:
```
2019-09-25 11:55:18,997 UTC [21341] INFO    Running 1 tasks using 1 processes
2019-09-25 11:55:19,016 UTC [27403] INFO    Starting task sit_anom/tas in process [27403]
2019-09-25 11:55:19,047 UTC [27403] DEBUG   Running block ['fix_metadata', 'concatenate', 'cmor_check_metadata', 'extract_time', 'fix_data', 'climate_statistics', 'cmor_check_data']
2019-09-25 11:55:19,047 UTC [27403] DEBUG   Applying single-model steps to PreprocessorFile: /scratch/b/b309059/esmvaltool_output/recipe_cmip3_test_20190925_115513/preproc/sit_anom/tas/CMIP3_NIES_miroc3_2_hires_A1_20c3m_run1_tas_1997-1999.nc
2019-09-25 11:55:19,048 UTC [27403] DEBUG   Running preprocessor step fix_file
2019-09-25 11:55:19,048 UTC [27403] DEBUG   Running fix_file(/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_hires/run1/tas_A1.nc, {'project': 'CMIP3', 'dataset': 'miroc3_2_hires', 'short_name': 'tas', 'output_dir': '/scratch/b/b309059/esmvaltool_output/recipe_cmip3_test_20190925_115513/preproc/sit_anom/tas/CMIP3_NIES_miroc3_2_hires_A1_20c3m_run1_tas_1997-1999_fixed'})
2019-09-25 11:55:19,049 UTC [27403] DEBUG   Running preprocessor step load
2019-09-25 11:55:19,049 UTC [27403] DEBUG   Running load(/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_hires/run1/tas_A1.nc, {'callback': <function concatenate_callback at 0x2b089cda6bf8>})
2019-09-25 11:55:19,049 UTC [27403] DEBUG   Loading:
/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_hires/run1/tas_A1.nc
2019-09-25 11:55:19,091 UTC [27403] DEBUG   Running preprocessor step fix_metadata
2019-09-25 11:55:19,091 UTC [27403] DEBUG   Running fix_metadata([<iris 'Cube' of air_temperature / (K) (time: 1212; latitude: 160; longitude: 320)>], {'project': 'CMIP3', 'dataset': 'miroc3_2_hires', 'short_name': 'tas', 'cmor_table': 'CMIP3', 'mip': 'A1', 'frequency': 'mo'})
2019-09-25 11:55:19,110 UTC [21341] INFO    Progress: 1 tasks running, 0 tasks waiting for ancestors, 0/1 done
2019-09-25 11:55:22,905 UTC [27403] DEBUG   Running preprocessor step concatenate
2019-09-25 11:55:22,906 UTC [27403] DEBUG   Running concatenate([<iris 'Cube' of air_temperature / (K) (time: 1212; latitude: 160; longitude: 320)>], {})
2019-09-25 11:55:22,907 UTC [27403] DEBUG   Running preprocessor step cmor_check_metadata
```
Should this step take longer?

**Comment by jvegasbsc** [comment]
> I think it is not that big problem but I can't use the institute table you defined in config_developer.yml.

This will give us trouble when trying to run the same recipes at diferent data centers

**Comment by jvegasbsc** [comment]
> Should this step take longer?

No, I was able to run the recipe quite fast. Never notice any slowdown on any file.

If you try to run again, the bahaviour is the same?

**Comment by LisaBock** [comment]
> If you try to run again, the bahaviour is the same?

Yes, I tried it now severel times and it started with different models but it always stopped there.

**Comment by jvegasbsc** [comment]
Ok. Can you put the file in a ftp or something, so I can try to debug it locally?

**Comment by LisaBock** [comment]
Here you could find two of our CMIP3 datasets for `tas`:

miroc3_2_hires: [tas_A1.nc](https://swift.dkrz.de/v1/dkrz_ee457bb86caa4970b691d7a96172026e/CMIP3/tas_A1.nc?temp_url_sig=0e4fec4ed8d4fcbb1a5c6eb9ddc6543e5ec1cafa&temp_url_expires=2019-10-02T12:29:14Z)

cccma_cgcm3_1: [tas_a1_20c3m_1_cgcm3.1_t47_1850_2000.nc](https://swift.dkrz.de/v1/dkrz_ee457bb86caa4970b691d7a96172026e/CMIP3/tas_a1_20c3m_1_cgcm3.1_t47_1850_2000.nc?temp_url_sig=51391f2c4bf647d59dff13f4efe8d3256b1b352f&temp_url_expires=2019-10-02T12:30:38Z)

**Comment by jvegasbsc** [comment]
Thanks!

**Comment by jvegasbsc** [comment]
I can not reproduce the error locally and I don't have access to DKRZ. Maybe you can try to get help from @mattiarighi or @schlunma. If they also have the same problem maybe we can collect enough info to learn what it is really happening.

**Comment by jvegasbsc** [comment]
As it is working except for this model issues, I think we can start the process to merge it

**Comment by LisaBock** [comment]
When  I set `max_parallel_tasks: 1`in the config file, it is also working on DKRZ!

File: _sidebar.rst.inc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (bouweandela) [inline _sidebar.rst.inc]: I don't think this file should be here

File: esmvalcore/_data_finder.py [hunk 2]
```diff
@@ -67,16 +69,35 @@ def get_start_end_year(filename):
...
     if len(dates) == 1:
         start_year = int(dates[0][:4])
         end_year = start_year
     elif len(dates) == 2:
         start_year, end_year = int(dates[0][:4]), int(dates[1][:4])
     else:
-        raise ValueError('Name {0} dates do not match a recognized '
-                         'pattern'.format(name))
-
+        # Slower than just parsing the name
```
Comment (bouweandela) [inline esmvalcore/_data_finder.py]: Is this needed? The preprocessor function `extract_time` will already take care of removing unnecessary data.

Comment (jvegasbsc) [reply esmvalcore/_data_finder.py]: Yes, but extract time is performed after several steps, including concatenation. If we don limit the reading to the really required files we can experiment errors before `extract_time` due to problems in files that we don't need at all

Comment (bouweandela) [reply esmvalcore/_data_finder.py]: If you expect such errors, I guess this is OK

File: _sidebar.rst.inc (hunk 1 above)
Comment (jvegasbsc) [reply _sidebar.rst.inc]: Removed and added to gitignore to avoid further problems

**Comment by jvegasbsc** [comment]
Done!

**Comment by mattiarighi** [comment]
I tried reading `tas` from the following dataset
```
 - {dataset: bccr_bcm2_0, project: CMIP3, mip: A1, modeling_realm: atm, exp: 20c3m, frequency: mo, ensemble: run1, start_year: 1980, end_year: 1999}
```
but `_data_finder.py` complains about another model...

```
2020-01-02 14:09:34,231 UTC [22573] ERROR   Program terminated abnormally, see stack trace below for more information
Traceback (most recent call last):
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_main.py", line 229, in run
    conf = main(args)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_main.py", line 157, in main
    process_recipe(recipe_file=recipe, config_user=cfg)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_main.py", line 203, in process_recipe
    recipe = read_recipe_file(recipe_file, config_user)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 72, in read_recipe_file
    recipe_file=filename)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 884, in __init__
    self.tasks = self.initialize_tasks() if initialize_tasks else None
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 1232, in initialize_tasks
    task_name=task_name,
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 855, in _get_preprocessor_task
    name=task_name,
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 713, in _get_single_preprocessor_task
    config_user=config_user,
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 674, in _get_preprocessor_products
    ancestors = _get_ancestors(variable, config_user)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 484, in _get_ancestors
    input_files = _get_input_files(variable, config_user)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 472, in _get_input_files
    drs=config_user['drs'])
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_data_finder.py", line 268, in get_input_filelist
    variable['end_year'])
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_data_finder.py", line 111, in select_files
    start, end = get_start_end_year(filename)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_data_finder.py", line 99, in get_start_end_year
    'not be readed from the file'.format(filename)
ValueError: File /mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/2xco2/atm/mo/tas/ncar_ccsm3_0/run1/tas_A1.2XCO2_1.CCSM.atmm.0000-01_cat_0067-12.nc dates do not match a recognized pattern and time can not be readed from the file
2020-01-02 14:09:34,235 UTC [22573] INFO    If you suspect this is a bug or need help, please open an issue on https://github.com/ESMValGroup/ESMValTool/issues and attach the run/recipe_*.yml and run/main_log_debug.txt files from the output directory.
```

File: esmvalcore/_data_finder.py [hunk 3]
```diff
@@ -67,16 +69,35 @@ def get_start_end_year(filename):
...
+            except iris.exceptions.CoordinateNotFoundError:
+                continue
+            start_year = time.cell(0).point.year
+            end_year = time.cell(-1).point.year
+            break
+
+    if not start_year or not end_year:
+        raise ValueError(
+            'File {0} dates do not match a recognized pattern and time can '
+            'not be readed from the file'.format(filename)
```
Comment (bouweandela) [inline esmvalcore/_data_finder.py]:
```suggestion
            'not be read from the file'.format(filename)
```

File: esmvalcore/_data_finder.py [hunk 4]
```diff
@@ -67,16 +69,35 @@ def get_start_end_year(filename):
...
+            logger.debug(cube)
+            try:
+                time = cube.coord('time')
+            except iris.exceptions.CoordinateNotFoundError:
+                continue
+            start_year = time.cell(0).point.year
+            end_year = time.cell(-1).point.year
+            break
+
+    if not start_year or not end_year:
```
Comment (bouweandela) [inline esmvalcore/_data_finder.py]:
```suggestion
    if start_year is None or end_year is None:
```
or this will also fail if the year is 0

**Comment by mattiarighi** [comment]
Tested again, with `bccr_bcm2_0` and `cccma_cgcm3_1`. In both cases, it hangs on the concatenation step of the preprocessor:

```
2020-01-15 09:15:32,185 UTC [4106] DEBUG   esmvalcore.preprocessor:208 Running preprocessor step concatenate
2020-01-15 09:15:32,189 UTC [4106] DEBUG   esmvalcore.preprocessor:198 Running concatenate([<iris 'Cube' of air_temperature / (K) (time: 1800; latitude: 60; longitude: 128)>, <iris 'Cube' of air_temperature / (K) (time: 1488; latitude: 46; longitude: 72)>, <iris 'Cube' of air_temperature / (K) (time: 1812; latitude: 60; longitude: 90)>, <iris 'Cube' of air_temperature / (K) (time: 1440; latitude: 46; longitude: 72)>, <iris 'Cube' of air_temperature / (K) (time: 1212; latitude: 160; longitude: 320)>, <iris 'Cube' of air_temperature / (K) (time: 600; latitude: 64; longitude: 128)>, <iris 'Cube' of air_temperature / (K) (time: 1812; latitude: 64; longitude: 128)>, <iris 'Cube' of air_temperature / (K) (time: 1812; latitude: 48; longitude: 96)>, <iris 'Cube' of air_temperature / (K) (time: 120; latitude: 128; longitude: 256)>, <iris 'Cube' of air_temperature / (K) (time: 1812; latitude: 64; longitude: 128)>, <iris 'Cube' of air_temperature / (K) (time: 1692; latitude: 72; longitude: 96)>, <iris 'Cube' of air_temperature / (K) (time: 1680; latitude: 145; longitude: 192)>, <iris 'Cube' of air_temperature / (K) (time: 1680; latitude: 73; longitude: 96)>, <iris 'Cube' of air_temperature / (K) (time: 1572; latitude: 160; longitude: 320)>, <iris 'Cube' of air_temperature / (K) (time: 1320; latitude: 64; longitude: 128)>, <iris 'Cube' of air_temperature / (K) (time: 2892; latitude: 96; longitude: 192)>, <iris 'Cube' of air_temperature / (K) (time: 1560; latitude: 96; longitude: 192)>, <iris 'Cube' of air_temperature / (K) (time: 1560; latitude: 45; longitude: 72)>, <iris 'Cube' of air_temperature / (K) (time: 1440; latitude: 46; longitude: 72)>], {})
```

This is strange, since for these models there is nothing to concatenate:
```
INFO    [46362] Using input files for variable tas of dataset cccma_cgcm3_1:
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/iap_fgoals1_0_g/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/giss_model_e_r/run1/tas_A1.GISS1.20C3M.run1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/giss_aom/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/giss_model_e_h/run1/tas_A1.GISS3.20C3M.run1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_hires/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/bccr_bcm2_0/run1/tas_A1_2.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/cccma_cgcm3_1_t63/run1/tas_a1_20c3m_1_cgcm3.1_t63_1850_2000.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/cccma_cgcm3_1/run1/tas_a1_20c3m_1_cgcm3.1_t47_1850_2000.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_ccsm3_0/run1/tas_A1.20C3M_1.CCSM.atmm.1990-01_cat_1999-12.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_medres/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ipsl_cm4/run1/tas_A1_1860-2000.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ukmo_hadgem1/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ukmo_hadcm3/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ingv_echam4/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_pcm1/run1/tas_A1.20C3M_1.PCM1.atmm.1890-01_cat_1999-12.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/mpi_echam5/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/csiro_mk3_0/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/inmcm3_0/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/2xco2/atm/mo/tas/giss_model_e_r/run1/tas_A1.GISS1.2xCO2.nc
```

**Comment by jvegasbsc** [comment]
From @LisaBock :
> When I set max_parallel_tasks: 1 in the config file, it is also working on DKRZ!

Are you using one or more processes?

**Comment by mattiarighi** [comment]
Sorry, I missed that comment.

Tried again with 1 processor and `bccr_bcm2_0`, still crashing on concatenate.

[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/4064231/main_log_debug.txt)

**Comment by jvegasbsc** [comment]
This looks like you have not set the drs properly in your config-user
```
Looking for files matching ['tas_*.nc'] in ['/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/']
```
And its finding every file for each dataset:
```
2020-01-15 09:34:26,392 UTC [20438] INFO    esmvalcore._recipe:517 Using input files for variable tas of dataset bccr_bcm2_0:
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/iap_fgoals1_0_g/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/giss_model_e_r/run1/tas_A1.GISS1.20C3M.run1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/giss_aom/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/giss_model_e_h/run1/tas_A1.GISS3.20C3M.run1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_hires/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/bccr_bcm2_0/run1/tas_A1_2.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/cccma_cgcm3_1_t63/run1/tas_a1_20c3m_1_cgcm3.1_t63_1850_2000.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/cccma_cgcm3_1/run1/tas_a1_20c3m_1_cgcm3.1_t47_1850_2000.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_ccsm3_0/run1/tas_A1.20C3M_1.CCSM.atmm.1990-01_cat_1999-12.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/miroc3_2_medres/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ipsl_cm4/run1/tas_A1_1860-2000.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ukmo_hadgem1/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ukmo_hadcm3/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ingv_echam4/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/ncar_pcm1/run1/tas_A1.20C3M_1.PCM1.atmm.1890-01_cat_1999-12.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/mpi_echam5/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/csiro_mk3_0/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/20c3m/atm/mo/tas/inmcm3_0/run1/tas_A1.nc
/mnt/lustre02/work/bd0854/DATA/ESMValTool2/CMIP3/2xco2/atm/mo/tas/giss_model_e_r/run1/tas_A1.GISS1.2xCO2.nc
```

**Comment by mattiarighi** [comment]
I'm collecting stupid mistakes today... :disappointed: 
It works now, thanks!

I'm going to open an issue for the 1-processor problem, we should not forget it.

**Comment by valeriupredoi** [comment]
also `...and time can not be readed...` -> is it a Spanish person who wrote the error message? :rofl:

**Pull request merged**